l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,363 @@
+        // `etc/env/ecosystems/ENV.yaml` file in `zksync-era` repository
+        bytes diamondCutData;
+        // Address of the governance contract for the L1 ecosystem
+        address governance;
+        // Address of the L1 shared bridge proxy
+        address l1SharedBridgeProxy;
+        // Address of the L1 ERC20 bridge proxy (required for the L2 bridge deployment)
+        address erc20BridgeProxy;
+    }
+
+    struct ContractsBytecodes {
+        // Default bytecode of the ERC-20 on L2 (BeaconProxy)
+        bytes beaconProxy;
+        // Bytecode of the L2 shared bridge
+        bytes l2SharedBridgeBytecode;
+        // Bytecode of the L2 shared bridge proxy (TransparentUpgradeableProxy)
+        bytes l2SharedBridgeProxyBytecode;
+    }
+
+    Config internal config;
+    ContractsBytecodes internal bytecodes;
+
+    function run() public {
+        console.log("Preparing ZK chain registration calldata");
+
+        initializeConfig();
+
+        checkBaseTokenAddress();
+
+        IGovernance.Call[] memory calls;
+        uint256 cnt = 0;
+        if (!IBridgehub(config.bridgehub).tokenIsRegistered(config.baseToken)) {
+            calls = new IGovernance.Call[](2);
+            console.log("Adding a call to register base token on the bridgehub");
+            IGovernance.Call memory baseTokenRegistrationCall = prepareRegisterBaseTokenCall();
+            calls[cnt] = baseTokenRegistrationCall;
+            ++cnt;
+        } else {
+            calls = new IGovernance.Call[](1);
+        }
+
+        IGovernance.Call memory registerChainCall = prepareRegisterHyperchainCall();
+        calls[cnt] = registerChainCall;
+        ++cnt;
+
+        address l2SharedBridgeProxy = computeL2BridgeAddress();
+        IGovernance.Call memory initChainCall = prepareInitializeChainGovernanceCall(l2SharedBridgeProxy);
+
+        scheduleTransparentCalldata(calls, initChainCall);
+    }
+
+    function initializeConfig() internal {
+        // Grab config from output of l1 deployment
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script-config/prepare-registration-calldata.toml");
+        console.log("Reading a config from", path);
+        string memory toml = vm.readFile(path);
+
+        // Config file must be parsed key by key, otherwise values returned
+        // are parsed alphabetically and not by key.
+        // https://book.getfoundry.sh/cheatcodes/parse-toml
+        config.stateTransitionProxy = toml.readAddress("$.deployed_addresses.state_transition_proxy_addr");
+        config.bridgehub = IStateTransitionManager(config.stateTransitionProxy).BRIDGE_HUB();
+        config.l1SharedBridgeProxy = address(Bridgehub(config.bridgehub).sharedBridge());
+        config.erc20BridgeProxy = toml.readAddress("$.deployed_addresses.erc20_bridge_proxy_addr");
+
+        config.chainId = toml.readUint("$.chain.chain_id");
+        config.eraChainId = toml.readUint("$.chain.era_chain_id");
```

Comment:
do we still need to load it from the config? could it simply be a constant?

Reply:
different envs - different values, plus the testnets are changed from time to time, same as our chain id there